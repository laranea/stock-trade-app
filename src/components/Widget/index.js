// @flow
import * as React from 'react';
import styled from 'styled-components';

const Container = styled.div``;

const Header = styled.div`
  padding: 0.625rem 1.25rem;
  background: #999999;
  border-radius: 0.625rem 0.625rem 0 0;
`;

const Body = styled.div`
  padding: 1.25rem;
  background: #ffffff;
  border-radius: ${props => (props.title ? '0 0 .625rem .625rem' : '.625rem')};
`;

const Title = styled.h4`
  margin: 0;
  color: #ffffff;
`;

type WidgetProps = {
  children: React.Node,
  title?: string
};

function Widget(props: WidgetProps) {
  return (
    <Container>
      {props.title && (
        <Header>
          <Title>{props.title}</Title>
        </Header>
      )}
      <Body title={props.title}>{props.children}</Body>
    </Container>
  );
}

export default Widget;
