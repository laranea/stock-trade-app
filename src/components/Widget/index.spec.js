import React from 'react';
import { mount } from 'enzyme';

import Widget from './';

describe('Widget component', () => {
  describe('When Widget title present', () => {
    it('should match snapshot', () => {
      const component = mount(
        <Widget>
          <h1>Hello World</h1>
        </Widget>
      );
      expect(component).toMatchSnapshot();
    });
  });

  describe('When Widget title not present', () => {
    it('should match snapshot', () => {
      const component = mount(
        <Widget title="TESTING">
          <h1>Hello World</h1>
        </Widget>
      );
      expect(component).toMatchSnapshot();
    });
  });
});
