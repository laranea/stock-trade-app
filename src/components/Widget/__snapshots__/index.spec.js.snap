// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget component When Widget title not present should match snapshot 1`] = `
<Widget
  title="TESTING"
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "bDWFJH",
            "rules": Array [],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa bDWFJH"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "kOvYal",
                  "rules": Array [
                    "
  padding: 0.625rem 1.25rem;
  background: #999999;
  border-radius: 0.625rem 0.625rem 0 0;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH kOvYal"
            >
              <styled.h4>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": true,
                        "lastClassName": "fwrZiM",
                        "rules": Array [
                          "
  margin: 0;
  color: #ffffff;
",
                        ],
                      },
                      "displayName": "styled.h4",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "h4",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <h4
                    className="sc-bxivhb fwrZiM"
                  >
                    TESTING
                  </h4>
                </StyledComponent>
              </styled.h4>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div
          title="TESTING"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "iQTiKG",
                  "rules": Array [
                    "
  padding: 1.25rem;
  background: #ffffff;
  border-radius: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            title="TESTING"
          >
            <div
              className="sc-htpNat iQTiKG"
              title="TESTING"
            >
              <h1>
                Hello World
              </h1>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Widget>
`;

exports[`Widget component When Widget title present should match snapshot 1`] = `
<Widget>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "bDWFJH",
            "rules": Array [],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa bDWFJH"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "cjPpoA",
                  "rules": Array [
                    "
  padding: 1.25rem;
  background: #ffffff;
  border-radius: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat cjPpoA"
            >
              <h1>
                Hello World
              </h1>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Widget>
`;
